src/systemd-sonic-generator/systemd-sonic-generator.c

```diff
@@ -547,8 +550,8 @@ int get_num_of_asic() {
         if (fp != NULL) {
             while ((nread = getline(&line, &len, fp)) != -1) {
                 if (strstr(line, "NUM_ASIC") != NULL) {
-                    token = strtok(line, "=");
-                    str_num_asic = strtok(NULL, "=");
+                    token = strtok_r(line, "=", &saveptr2);
```

Comment:
can we use the same saveptr? instead of two separate pointers?

Reply:
I think we can. Fixed to one pointer.